### PR TITLE
Allow regex checks with yamlfile_value template, update add_platform_rule.py

### PR DESCRIPTION
**Pull request by mrogers950**
This PR adds pattern_check input to the template, which gives you a pattern match operation type. It also fixes the helper utility to accommodate recent changes and to begin using the template, with an added --regex flag to set pattern_check in the rule definition. I can split these up into different PR's if that's preferred.

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python abstract syntax tree change found in ssg/templates.py.
 Python abstract syntax tree change found in utils/add_platform_rule.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python abstract syntax tree change found in ssg/templates.py.
 Python abstract syntax tree change found in utils/add_platform_rule.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python abstract syntax tree change found in ssg/templates.py.
 Python abstract syntax tree change found in utils/add_platform_rule.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

File: utils/add_platform_rule.py [hunk 1]
```diff
@@ -78,58 +86,25 @@
...
-
-   <external_variable comment="Root of downloaded stuff" datatype="string" id="ocp_data_root" version="1" />
-</def-group>
-{{{{% endif  %}}}}
-''')
+def check_value(value):
+    if value:
+        return 'pattern_check: "true"\n        type: "string"'
+    else:
+        return 'entity_check: "at least one"'
```
Comment (yuumasato) [inline utils/add_platform_rule.py:106]: Why are `pattern_check` and `entity_check` mutually exclusive?
This means that when doing a `pattern match` all values found need to match.

Can there be use case when you want a `pattern match` but the test is fine if `at least one` value matches?

Comment (mrogers950) [reply utils/add_platform_rule.py:106]: @yuumasato oh yes, I misunderstood the use of entity_check initially. I'll fix that.

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python abstract syntax tree change found in ssg/templates.py.
 Python abstract syntax tree change found in utils/add_platform_rule.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by yuumasato** [comment]
Errors on CI seem unrelated.

**Pull request merged**